desc/objectives/_generic.py

```diff
@@ -15,6 +19,262 @@
+        self._eq = eq.copy()
+        self._fun = fun
+        self._dim_f = dim_f
+        self._fd_step = fd_step
+        self._vectorized = vectorized
+        self._kwargs = kwargs
+        if self._vectorized:
+            try:  # spawn a new environment so the backend can be set to numpy
+                mp.set_start_method("spawn")
+            except RuntimeError:  # context can only be set once
+                pass
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode="fwd",
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        self._scalar = self._dim_f == 1
+        self._constants = {"quad_weights": 1.0}
+
+        if self._vectorized:
+            max_processes = (
+                self._vectorized
+                if isinstance(self._vectorized, int)
+                else os.cpu_count()
+            )
+            self._pool = mp.Pool(processes=max_processes)
+
+        def fun_wrapped(params):
+            """Wrap external function with possibly vectorized params."""
+            # number of equilibria for vectorized computations
+            param_shape = params["Psi"].shape
+            num_eq = param_shape[0] if len(param_shape) > 1 else 1
+
+            if self._vectorized and num_eq > 1:
+                # convert params to list of equilibria
+                eqs = [self._eq.copy() for _ in range(num_eq)]
+                for k, eq in enumerate(eqs):
+                    # update equilibria with new params
+                    for param_key in self._eq.optimizable_params:
+                        param_value = np.array(params[param_key][k, :])
+                        if len(param_value):
+                            setattr(eq, param_key, param_value)
+                # parallelize calls to external function
+                max_processes = (
+                    self._vectorized
+                    if isinstance(self._vectorized, int)
+                    else os.cpu_count()
+                )
+                with mp.Pool(processes=min(max_processes, num_eq)) as pool:
+                    results = pool.map(
+                        functools.partial(self._fun, **self._kwargs), eqs
+                    )
+                    pool.close()
+                    pool.join()
+                    return jnp.vstack(results, dtype=float)
+            else:  # no vectorization
+                # update equilibrium with new params
+                for param_key in self._eq.optimizable_params:
```

Comment:
can't this just be `eq.params_dict = params`?

Reply:
No, because when vectorized `params` is a dict of 2D arrays with each row the params for a different equilibrium